### PR TITLE
Docs: updates request flow diagram

### PR DESCRIPTION
**Pull request by ZPain8464**
I honestly think the sequence diagram is the best approach to explain interactions between components. I revamped the current one so it's easier to read (I added color-coded services and more context). I also include 3 steps in the diagram that are explained beneath it, but they are somewhat repetitive if you read the Components section as well. 

Another thing worth noting is the "miro" watermark in the bottom-right corner. I can update to a free trial to see if I can save the image in higher resolution without the watermark for the time being. If everyone is OK with this new diagram, I'll upgrade and replace the current image. 

Tagging @desimone, @calebdoxsey, and @wasaga for visibility.

### CODE REVIEW

**Review by calebdoxsey — APPROVED** [review]
LGTM

**Review by wasaga — APPROVED** [review]

File: content/docs/internals/architecture.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ---
+# cspell:ignore Authorization
```
Comment (desimone) [inline content/docs/internals/architecture.md:2]:
```suggestion
# cspell:ignore Authorization
```

Probably should just add tthis to `cspell.json` instead?

File: content/docs/internals/architecture.md [hunk 2]
```diff
@@ -50,10 +51,30 @@ In test deployments, all four components may run from a [single binary and confi
...
-## Authentication Flow
+## The lifecycle of a request
 
-Pomerium's internal and external component interactions during full authentication from a fresh user are diagramed below.
+The diagram below shows how Pomerium's components communicate when authenticating a user.
 
 After initial authentication to provide a session token, only the authorization check interactions occur.
 
-![pomerium architecture diagram](./img/architecture/pomerium-auth-flow-stateless-auth.svg)
+![pomerium architecture diagram](./img/architecture/pomerium-request-flow-2.jpg)
```
Comment (desimone) [inline content/docs/internals/architecture.md:60]:
```suggestion
![pomerium architecture diagram](./img/architecture/pomerium-request-flow-2.jpg)
```

SVG would probably be better if you can export it.

File: content/docs/internals/architecture.md [hunk 3]
```diff
@@ -50,10 +51,30 @@ In test deployments, all four components may run from a [single binary and confi
...
+
+**Step 1:** Unauthenticated client
+
+The client sends a request to access a secured application, which prompts the Proxy to send the request over a gRPC call to the Authorization service.
+
+Because the client has not authenticated, the Authorization service can't locate the session ID in the Databroker service. It sends the client to the Authenticate service, which prompts the client to sign in to the IdP.
+
+**Step 2:** Authenticating the client
+
+After the client signs in, the IdP exchanges an authorization code with the Authentication service. The Authentication service uses the code to get OAuth tokens and OIDC claims from the IdP.
```
Comment (desimone) [inline content/docs/internals/architecture.md:70]:
```suggestion
After the client signs in, the IdP exchanges an authorization code with the Authentication service. The Authentication service uses the code to get OAuth tokens and OIDC claims from the IdP.
```

> authorization code
we should probably link to the oidc spec which describes code flow.

**Review by desimone — CHANGES_REQUESTED** [review]
There are two flow diagrams but only one is used. Also, SVG or a high quality png would be better than jpeg i think.

**Review by desimone — APPROVED** [review]

**Pull request merged**